File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        if (s.length % 2 !== 0) return false;
	    let dict = {	
		    '}': '{',
		    ']': '[',
		    ')': '(',
		    }
	    let stack = [];
	    for (let i = 0; i < s.length; i++) {
		    if (dict[s[i]] === undefined) {
	            stack.push(s[i]);
            } else {
	            if (dict[s[i]] !== stack[stack.length - 1]) return false;
                stack.pop();
            }
	    }
	    if (stack.length !== 0) return false;
	    return true;
    }
}

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        t = list(t)
        t.reverse()

        for c in s:
            if len(t) == 0:
                return 0
            
            if c == t[-1]:
                t.pop()
            
        return len(t)